src/library.js

```diff
@@ -4540,8 +4540,8 @@ LibraryManager.library = {
 #if ASSERTIONS
     // Nobody should have mutated _readAsmConstArgsArray underneath us to be something else than an array.
     assert(Array.isArray( __readAsmConstArgsArray));
-    // The input buffer is allocated on the stack, so it must be stack-aligned.
-    assert(buf % {{{ STACK_ALIGN }}} == 0);
```

Comment:
The old code here is certainly correct for our current use - but it does not reflect a required internal condition of this function, so I think it is not relevant here. The idea with that assert was to check that the loop below is safe to do the `buf >>= 2` trick, and not to assert stack alignment. Whether input is 4 or 8 byte aligned does not matter.